Psychtoolbox/PsychProbability/UniqueAndN.m

```diff
@@ -1,16 +1,27 @@
-function [b,n] = UniqueAndN(in)
-% [uniques,n] = uniqueen(input)
+function [b,n] = UniqueAndN(in,qByRow)
+% [uniques,n] = uniqueen(input,qByRow)
 %
 % returns the unique elements in input (sorted) and
 % the number of times the item ocurred
+% optionally does so for unique rows (qByRow=true) instead of unique
+% elements
 % 
 % DN    2008
+% DN    2017 Added byRow option
 
-in      = sort(in);  % Necessary for the trick below to work
-
-[b,i,j] = unique(in);
+if nargin<2 || ~qByRow

```

Comment:
Maybe also || isempty(qByRow) || ~qByRow, as otherwise an [] one will evaluate to true.